lib/errors.js

```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+const codes = {
```

Comment:
These should be actual codes, not strings. Look at how core is doing this:

https://github.com/nodejs/node/blob/a1fc52884b98acab3d3f0b66c3b5ac9eb0728b8b/lib/internal/errors.js#L704

https://github.com/nodejs/node/blob/a1fc52884b98acab3d3f0b66c3b5ac9eb0728b8b/lib/internal/errors.js#L227

Reply:
You mean native Errors objects? This is debatable. We already work with error objects and don't use inheritance.

Reply:
No, I mean `FST_ERR_FOO` should be equal to `42`.

Reply:
From where do you get this information that an error must be an integer? Node.js also using strings. [Here](https://github.com/nodejs/node/blob/a1fc52884b98acab3d3f0b66c3b5ac9eb0728b8b/lib/internal/errors.js#L227) `sym` is the code string and assigned as `code` property to the error object [here](https://github.com/nodejs/node/blob/a1fc52884b98acab3d3f0b66c3b5ac9eb0728b8b/lib/internal/errors.js#L211)

Reply:
Regardless, the issue is that strings are fragile. Symbols are fine.

Reply:
Yep, symbols are unique but can't be converting to strings (logging). Due to the fact that we prefix all strings we can guarantee their uniqueness.

Reply:
I think strings are fine, and what Node core is using. I’ll do a more detailed review asap.

Reply:
Mistake. Symbols are unique by default with a prefix we only guarantee collision-freedom. Do you think it will outperform the logging capabilities rather to write tests for that case?